sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -28,9 +28,16 @@ function isReadableStream(body: any): body is NodeJS.ReadableStream {
 
 function isStreamComplete(stream: NodeJS.ReadableStream): Promise<void> {
   return new Promise((resolve) => {
-    stream.on("close", resolve);
-    stream.on("end", resolve);
-    stream.on("error", resolve);
+    const handler = (): void => {
```

Comment:
This would not fix the original issue. Not sure we could add test cases for this warning. But to simplify on how to reproduce this we could leverage [one test case](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/translation/ai-translation-document-rest/test/public/node/documentFilterTest.spec.ts#L212) in translation document.
- cd translation/ai-translation-document-rest
- only that test case
- rush build -t @azure-rest/ai-translation-document && rushx unit-test:node
![image](https://github.com/user-attachments/assets/9952718f-a959-4f25-9db2-95f8a7cc2558)